src/OSPSuite.InstallationValidator.Core/Services/OutputComparisonResultToCurveChartMapper.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.Generic;
+using System.Drawing;
+using OSPSuite.Core.Chart;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Data;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Core.Extensions;
+using OSPSuite.InstallationValidator.Core.Domain;
+using OSPSuite.Utility;
+
+namespace OSPSuite.InstallationValidator.Core.Services
+{
+   public interface IOutputComparisonResultToCurveChartMapper : IMapper<OutputComparisonResult, IEnumerable<CurveChart>>
+   {
+   }
+
+   public class OutputComparisonResultToCurveChartMapper : IOutputComparisonResultToCurveChartMapper
+   {
+      private readonly IOutputResultToDataRepositoryMapper _outputResultToDataRepositoryMapper;
+      private readonly IDimensionFactory _dimensionFactory;
+
+      public OutputComparisonResultToCurveChartMapper(IOutputResultToDataRepositoryMapper outputResultToDataRepositoryMapper, IDimensionFactory dimensionFactory)
+      {
+         _outputResultToDataRepositoryMapper = outputResultToDataRepositoryMapper;
+         _dimensionFactory = dimensionFactory;
+      }
+
+      public IEnumerable<CurveChart> MapFrom(OutputComparisonResult outputComparisonResult)
+      {
+         var dataRepository1 = _outputResultToDataRepositoryMapper.MapFrom(outputComparisonResult.Output1);
+         var dataRepository2 = _outputResultToDataRepositoryMapper.MapFrom(outputComparisonResult.Output2);
+
+         var linearCurveChart = new CurveChart
```

Comment:
That looks like a method createChart  with (lin/log) as parameter could be defined. Code is exactly the same